### PR TITLE
Add autocomplete to text inputs - by exact form, by lemma

### PR DESCRIPTION
**Pull request by irina060981**
For the issue - https://github.com/alpheios-project/components/issues/562

1. Changed the list of filters:
image

2. Added autocomplete feature to text inputs - by exact forms and by lemma
image

3. Filtering is executed only with valid values

4. Filter block is hidden if there is only one item in the list
image

5. Filters are cleared after adding a new word to the wordlist

### CODE REVIEW

File: src/vue/components/word-list/word-language-panel.vue [hunk 1]
```diff
@@ -118,9 +126,34 @@ export default {
         } else {
           console.warn(`The current filter method - ${this.selectedFilterBy} - is not defined, that's why empty result is returned!`)
         }
-      }
+      }      
       return []
     },
+    wordExactForms () {
+      let exactForms = []
+      this.wordlist.values.map(wordItem => {
```
Comment (kirlat) [inline src/vue/components/word-list/word-language-panel.vue]: I think would be slightly better to use `reduce` for that. The following code will probably be simpler to read, on my opinion:
```
let exactForms = this.wordlist.values.reduce((acc, wordItem) => {
    let exactForm = wordItem.targetWord.toLowerCase()
    if (exactForms.includes(exactForm)) {
      acc.push(exactForm)
   }
}, [])
```

File: src/vue/components/word-list/word-language-panel.vue [hunk 2]
```diff
@@ -118,9 +126,34 @@ export default {
...
+        if (exactForms.indexOf(exactForm) === -1) {
+          exactForms.push(exactForm)
+        }
+      })
+      exactForms.sort()
+      return exactForms
+    },
+    wordLemmaForms () {
+      let lemmaForms = []
+      this.wordlist.values.map(wordItem => {
```
Comment (kirlat) [inline src/vue/components/word-list/word-language-panel.vue]: Probably reduce will be cleaner here, too

File: src/vue/components/word-list/word-filter-panel.vue [hunk 3]
```diff
@@ -63,48 +89,88 @@
...
           this.setClickedLemmaFilter()
         }
         return true
+      },
+      wordExactFormsFiltered () {
+        if (this.selectedFilterBy === 'byExactForm') {
+          if (this.textInput && this.textInput.length > 0) {
+            return this.wordExactForms.filter(exactForm => exactForm.indexOf(this.textInput) > -1).map(exactForm => {
+              let startIndex = exactForm.indexOf(this.textInput)
+              return exactForm.substr(0, startIndex) + '<span class="alpheios-select-input-filter-part">' + this.textInput + '</span>' + exactForm.substr(startIndex + this.textInput.length)
```
[the remark this answers is not in the document]
Comment (irina060981) [reply src/vue/components/word-list/word-filter-panel.vue]: Oh, that is a good suggestion! Will do that!

File: src/vue/components/word-list/word-language-panel.vue (hunk 1 above)
Comment (irina060981) [reply src/vue/components/word-list/word-language-panel.vue]: Fixed

File: src/vue/components/word-list/word-language-panel.vue (hunk 2 above)
Comment (irina060981) [reply src/vue/components/word-list/word-language-panel.vue]: Fixed

**Comment by irina060981** [comment]
@kirlat , as I fixed all your requirements - I will merge it to master!

**Pull request merged**